include/xgboost/data.h

```diff
@@ -258,14 +257,7 @@ class SparsePage {
   inline Inst operator[](size_t i) const {
     const auto& data_vec = data.HostVector();
     const auto& offset_vec = offset.HostVector();
-    size_t size;
-    // in distributed mode, some partitions may not get any instance for a feature. Therefore
-    // we should set the size as zero
-    if (rabit::IsDistributed() && i + 1 >= offset_vec.size()) {
-      size = 0;
```

Comment:
??? This logic is not necessary any more?

Reply:
There's a test added in https://github.com/dmlc/xgboost/pull/4045/files and I passed it.  I remember fixing something similar to this by forcing passing correct number of features from jvm package.  It was with the help of @wbo4958 .